app/modules/queues/helpers/localQueue.ts

```diff
@@ -51,20 +51,26 @@ export default class LocalQueue {
       }
     }) as { data: Job };
 
-    // Return job data with BullMQ-compatible methods
-    return this.addJobMethods(jobObject.data);
+    return this.addBullMQCompatibility(jobObject.data);
 
   }
 
-  private getJobs = async (match: any) => {
+  private getJobs = async (match: any, skip?: number, limit?: number) => {
     const documents = getDocumentsAdapter();
 
     match.queue = this.name;
 
-    return await documents.getDocuments({
+    const result = await documents.getDocuments({
       collection: 'jobs',
-      match
+      match,
+      ...(skip !== undefined && { skip }),
+      ...(limit !== undefined && { limit })
     }) as { data: Job[], count: number };
+
+    return {
+      ...result,
```

Comment:
Why is this spread and also set as data?
Also, if it's returning data but it's actually an object now, should this be changed just to return either a jobs: [] or just an array?